test/inductor/test_codecache.py

```diff
@@ -37,3 +53,245 @@ def test_codecache_spawn():
 @requires_cuda()
 def test_codecache_fork():
     _run_codecache_test("fork")
+
+
+@instantiate_parametrized_tests
+class TestFxGraphCache(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Reroute all cache disk activity to a clean temporary directory to
+        # ensure isolation (and initial cache misses). Deliberately create the
+        # temp dir in setUpClass, however, so that individual test runs reuse
+        # the same location. We don't expect different tests to reuse cache
+        # entries, so preserving the temp dir provides that additional testing.
+        cls.tmpdir = tempfile.TemporaryDirectory()
+        cls.cache_dir_patch = patch("torch._inductor.codecache.cache_dir")
+        cls.cache_dir_patch.start().return_value = cls.tmpdir.name
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.cache_dir_patch.stop()
+        cls.tmpdir.cleanup()
+
+    @config.patch({"fx_graph_cache": True})
+    @parametrize("device", ("cuda", "cpu"))
+    @parametrize("dtype", (torch.float, torch.bfloat16))
+    def test_cache_load(self, device, dtype):
+        """
+        Verify that we can populate and load CompiledFxGraphs from the cache.
+        """
+        if device == "cuda" and not HAS_CUDA:
+            raise unittest.SkipTest("requires CUDA")
+
+        def fn(x, y):
+            return (x * 2, y @ y)
+
+        a = torch.rand([25], dtype=dtype, device=device)
+        b = torch.rand([5, 5], dtype=dtype, device=device)
+        c = a.view(5, 5)
+
+        # Spy on these methods so we can verify they are called.
+        mock_save_graph = Mock(wraps=FxGraphCache.save_graph)
+        mock_load_graph = Mock(wraps=FxGraphCache.load_graph)
+        FxGraphCache.save_graph = mock_save_graph
+        FxGraphCache.load_graph = mock_load_graph
+
+        compiled_fn = torch.compile(fn, dynamic=False)
+
+        # A first call shold miss in the cache.
+        self.assertEqual(fn(a, b), compiled_fn(a, b))
+        self.assertEqual(mock_save_graph.call_count, 1)
+        self.assertEqual(mock_load_graph.call_count, 0)
+
+        # A second call should hit. (First reset so in-memory guards
+        # don't prevent compilation).
+        torch._dynamo.reset()
+        self.assertEqual(fn(a, b), compiled_fn(a, b))
+        self.assertEqual(mock_save_graph.call_count, 1)
+        self.assertEqual(mock_load_graph.call_count, 1)
+
+        # But we expect different code if the tensors are aliased.
+        torch._dynamo.reset()
+        self.assertEqual(fn(a, c), compiled_fn(a, c))
+        self.assertEqual(mock_save_graph.call_count, 2)
+        self.assertEqual(mock_load_graph.call_count, 1)
+
+    def test_tensor_constants(self):
+        """
+        Test the handling of small vs. large tensor constants.
+        """
+        data = FxGraphCachePickler.dumps(torch.tensor([i for i in range(9)]))
```

Comment:
Maybe could you add some that explicitly invoke inductor ? 

e.g. with something like

```
        def fn():
            return (torch.tensor([i for i in range(8)], device="cuda") + 10,)

        fn_fx = make_fx(fn)()
        fn_compiled = compile_fx_inner(fn_fx, [])
```

and changing the generated constant and see if we do a new recompilation

Reply:
Ok, for my education why do you suggest to explicitly invoke inductor for this particular test vs. a torch.compile?